### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by Spuds** [elkarte/Elkarte#3430]
The new notifications area seems to be in flux, and I just can't figure out what direction you were going. 

- If you go to the profile notifications page, ```template_action_notification``` it does not pass anything back (its missing the name attribute on the select), and also I think it should be using the 'status' for default checkbox and 'user' for selected options pulldown. I "fixed" that to at least send the values back
- I then found myself in ```makeNotificationChanges``` which is not setup to handle the arrays being sent back from the multiselect form.  I kind of fixed that based on if they choose default or individual options.  If they chose default they get the ACP default items for that mention.  At that point I had arrays such as  ["likemsg"]=> array(2) { ["email"]=> int(1) notification"]=> int(0) }.  A mention array with values set for the allowed notification methods.
- Then onto ```saveUserNotificationsPreferences``` OK no idea what data to save, maybe serialized array into the notification type field, but its probably not big enough.

So I'm tapping out on this one since I'm not clear what direction you were taking here and I'm probably making things more complicated then needed.  HELP! 🆘

**Comment by Spuds** [comment]
Forgot one thing Mr $_POSTman 🏣 😝

**Comment by emanuele45** [comment]
Arg!
I see what I've done. (Basically I thought I had completed it, but in fact I changed my mind mid-way and I did not fix all the posts I should have. :laughing: )
I'm going through it.
The idea is:
* `{db_prefix}notifications_pref` stores the settings in the form | id_member | mention_type | notification_type |
* if `notification_type` is set to "none" then it's deselected (i.e. no notification, not even the default set by the admin)
* if no row is present, but a default is set in $modSettings['notification_methods'] (i.e. value = Notifications::DEFAULT_LEVEL) then the default is used
* on saving: everything is deleted (and this is broken too) and new settings are re-INSERT'ed
* on reading: what's in `{db_prefix}notifications_pref` is integrated with the defaults

While working on this code I went back and forth with notification_type and notification_level, so I made a mess...
